avocado/core/multiplexer.py

```diff
@@ -377,28 +377,67 @@ def iteritems(self):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
+def _report_already_parsed(self, *args, **kwargs):
+    """
+    Raises exception describing that `self.data` alteration is restricted
+    """
+    raise RuntimeError("Altering of `self.data` is restricted. %s %s"
```

Comment:
"Altering" is indeed a valid word, but one that is very seldom seen (and possibly easily understood).  "Modifying" would be the more mundane (and boring?) version.